runtime/jcl/common/mgmtthread.c

```diff
@@ -937,14 +942,19 @@ getThread(JNIEnv *env, jlong threadID)
 	currentThread = javaVM->mainThread;
 	/* Loop over all vmThreads until we get back to the starting thread: look for matching threadID */
 	while (TRUE) {
-		if ((currentThread->threadObject != NULL) && (getThreadID((J9VMThread *)env, (j9object_t)currentThread->threadObject)  == threadID)) {
+#if JAVA_SPEC_VERSION >= 19
+		j9object_t threadObject = currentThread->carrierThreadObject;
+#else /* JAVA_SPEC_VERSION >= 19 */
+		j9object_t threadObject = currentThread->threadObject;
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		if ((NULL != threadObject) && (threadID == getThreadID((J9VMThread *)env, threadObject))) {
 			{
 				/*
 				 * We've found our matching thread, so we're done.
 				 * But first we have to check if the thread is alive
 				 * i.e. The j.l.Thread object's thread ref != 0
 				 */
-				if (J9VMJAVALANGTHREAD_THREADREF((J9VMThread *)env,currentThread->threadObject) == NULL) {
+				if (NULL == J9VMJAVALANGTHREAD_THREADREF((J9VMThread *)env, threadObject)) {
```

Comment:
need to be careful here as the currentThread returned is could be the virtualthread stack, the caller of `getThread` will have to verify that no virtual thread/continuation is mounted on top or caller will have to create a stack allocated thread containing the J9VMContinuation data